### PR TITLE
DAOS-6626 placement: fix memory leak

### PR DESCRIPTION
**Pull request by dmiter**
add_layout is not free when exit from the function.

### CODE REVIEW

**Comment by dmiter** [comment]
Don't know what means: "Some quality gates have been missed: overall result is FAILED" in NLT testing.

**Comment by ashleypittman** [comment]
> Don't know what means: "Some quality gates have been missed: overall result is FAILED" in NLT testing.

This was a recent change, we fail the build if there are any assertion failures logged.  It would go away if you rebased, you can you land this PR as is.

**Comment by ashleypittman** [comment]
I've got a seperate fix for jump_map.c in https://github.com/daos-stack/daos/pull/3902/files

**Comment by dmiter** [comment]
Ok. In PR#3902 just added correct error handling instead of assert.

File: src/placement/jump_map.c [hunk 1]
```diff
@@ -870,7 +870,7 @@ jump_map_obj_place(struct pl_map *map, struct daos_obj_md *md,
 	/* Allocate space to hold the layout */
 	rc = pl_obj_layout_alloc(jmop.jmop_grp_size, jmop.jmop_grp_nr,
 				 &layout);
-	if (rc != 0) {
+	if (rc) {
```
Comment (jolivier23) [inline src/placement/jump_map.c:873]: minor, I prefer the != 0 though the effect is the same.  It's not a boolean variable

**Review by jolivier23 — APPROVED** [review]

**Review by ashleypittman — APPROVED** [review]

**Comment by dmiter** [comment]
CartNoPmixOneNodeTest - fail not related to this patch.

File: src/placement/jump_map.c [hunk 2]
```diff
@@ -898,27 +903,25 @@ jump_map_obj_place(struct pl_map *map, struct daos_obj_md *md,
 		D_INIT_LIST_HEAD(&add_list);
 		layout_find_diff(jmap, layout, add_layout, &add_list);
 
-		if (!d_list_empty(&add_list)) {
-
+		if (!d_list_empty(&add_list))
 			rc = pl_map_extend(layout, &add_list, true);
-			if (rc != 0)
-				return rc;
```
Comment (gnailzenh) [inline src/placement/jump_map.c:905]: might change it to "goto out", just in case we add more code after the "if" statement

**Review by gnailzenh — APPROVED** [review]

**Pull request merged**